drivers/vmwarevsphere/vm.go

```diff
@@ -158,45 +159,53 @@ func (d *Driver) provisionVm(vm *object.VirtualMachine) error {
 	flag := FileAttrFlag{}
 	flag.SetPerms(0, 0, 660)
 
-	tmpDir, err := fileman.CreateTemporaryDirectory(d.getCtx(), auth.Auth(), "docker_", "", "/tmp")
-	if err != nil {
-		return err
-	}
+	if strings.ToLower(d.MachineOS) == defaultMachineOS {
```

Comment:
i think i'd like this better re-written as a switch with a default case to error out instead of two if statements

Reply:
done